services-core/src/economic_viability.rs

```diff
@@ -26,26 +26,29 @@ impl EconomicViabilityStrategy {
         &self,
         subsidy_factor: f64,
         min_avg_fee_factor: f64,
-        fallback_min_avg_fee_per_order: u128,
-        fallback_max_gas_price: u128,
+        static_min_avg_fee_per_order: Option<u128>,
+        static_max_gas_price: Option<u128>,
         native_token_price: Arc<dyn NativeTokenPricing + Send + Sync>,
         gas_station: Arc<dyn GasPriceEstimating + Send + Sync>,
-    ) -> impl EconomicViabilityComputing {
-        let mut economic_viabilities = Vec::<Box<dyn EconomicViabilityComputing>>::new();
-        if matches!(self, EconomicViabilityStrategy::Dynamic) {
-            economic_viabilities.push(Box::new(EconomicViabilityComputer::new(
+    ) -> Result<Arc<dyn EconomicViabilityComputing>> {
+        Ok(match self {
+            Self::Dynamic => Arc::new(EconomicViabilityComputer::new(
```

Comment:
Should we also assert here that static_max_gas_price and static_min_avg_fee_per_order are not set to avoid semantically undefined config state?

Reply:
That sounds useful, added it.